### PR TITLE
Feature/non jwt apis

### PR DESCRIPTION
**Pull request by annashamray**
Changes:
1. add support for APIs with token auth
1. add support for APIs with custom OAS url

### CODE REVIEW

File: zgw_consumers/models.py [hunk 1]
```diff
@@ -26,6 +29,20 @@ class Service(models.Model):
...
+    auth_type = models.CharField(
+        _("authorization type"),
+        max_length=20,
+        choices=AuthTypes,
+        default=AuthTypes.jwt,
+    )
+    header_key = models.CharField(_("header key"), max_length=100, blank=True)
+    header_value = models.CharField(_("header value"), max_length=255, blank=True)
+    oas = models.URLField(
+        _("OAS"), max_length=255, help_text=_("Path to OAS yaml file")
```
Comment (sergei-maertens) [inline zgw_consumers/models.py]:
```suggestion
        _("OAS"), max_length=1000, help_text=_("URL to OAS yaml file")
```

File: zgw_consumers/models.py [hunk 2]
```diff
@@ -26,6 +29,20 @@ class Service(models.Model):
...
+    )
+    nlx = models.URLField(
+        _("NLX url"), max_length=255, blank=True, help_text=_("NLX inway address")
```
Comment (sergei-maertens) [inline zgw_consumers/models.py]:
```suggestion
        _("NLX url"), max_length=1000, blank=True, help_text=_("NLX (outway) address")
```

File: zgw_consumers/constants.py [hunk 3]
```diff
@@ -14,3 +14,9 @@ class APITypes(DjangoChoices):
     kic = ChoiceItem("kic", _("KIC (Klantinteracties)"))
 
     orc = ChoiceItem("orc", _("ORC (Overige)"))
+
+
+class AuthTypes(DjangoChoices):
+    no_auth = ChoiceItem("no_auth", _("No authorization"))
+    token = ChoiceItem("token", _("Token"))
```
Comment (sergei-maertens) [inline zgw_consumers/constants.py]: I'd name this `API key` rather than token

File: zgw_consumers/constants.py [hunk 4]
```diff
@@ -14,3 +14,9 @@ class APITypes(DjangoChoices):
...
+    jwt = ChoiceItem("jwt", _("JWT"))
```
Comment (sergei-maertens) [inline zgw_consumers/constants.py]: I'd name this `ZGW Client ID + secret` rather. JWT has many different forms and is more of a mechanism.

File: zgw_consumers/models.py [hunk 5]
```diff
@@ -37,6 +54,13 @@ def __str__(self):
...
+
+        if not self.nlx.endswith("/"):
+            self.nlx = f"{self.nlx}/"
+
+        if not self.oas:
+            self.oas = urljoin(self.api_root, "schema/openapi.yaml")
+
         super().save(*args, **kwargs)
 
     def clean(self):
```
Comment (sergei-maertens) [inline zgw_consumers/models.py]: I'd validate that if `header_key` or `header_value` is set, the other one must also be set.

File: zgw_consumers/models.py [hunk 6]
```diff
@@ -68,5 +92,46 @@ def build_client(self, **claims):
...
+        if self.auth_type == AuthTypes.jwt:
+            client.auth = ClientAuth(
+                client_id=self.client_id, secret=self.secret, **claims
+            )
+        elif self.auth_type == AuthTypes.token:
+            client.auth_value = {self.header_key: self.header_value}
         return client
+
+    @classmethod
+    def get_service(cls, url: str):
```
Comment (sergei-maertens) [inline zgw_consumers/models.py]:
```suggestion
    def get_service(cls, url: str) -> "Service":
```

File: zgw_consumers/client.py [hunk 7]
```diff
@@ -1,8 +1,50 @@
...
-    client_class = getattr(settings, "ZGW_CONSUMERS_CLIENT_CLASS", "zds_client.Client")
+    client_class = getattr(
+        settings, "ZGW_CONSUMERS_CLIENT_CLASS", "zgw_consumers.client.ZGWClient"
+    )
     Client = import_string(client_class)
     return Client
+
+
+class ZGWClient(Client):
+    auth_value = None
```
Comment (sergei-maertens) [inline zgw_consumers/client.py]:
```suggestion
    auth_value: Optional[Dict[str, str]] = None
```

File: zgw_consumers/client.py [hunk 8]
```diff
@@ -1,8 +1,50 @@
...
+    schema_url = None
```
Comment (sergei-maertens) [inline zgw_consumers/client.py]:
```suggestion
    schema_url: str = ""
```

File: zgw_consumers/constants.py (hunk 3 above)
Comment (annashamray) [reply zgw_consumers/constants.py]: renamed to `api_key`

File: zgw_consumers/constants.py (hunk 4 above)
Comment (annashamray) [reply zgw_consumers/constants.py]: renamed to `zgw`

**Comment by sergei-maertens** [comment]
zgw-consumers==0.7.0 is on PyPI

**Pull request merged**